exonum-java-binding/integration-tests/src/test/java/com/exonum/binding/test/BlockchainIntegrationTest.java

```diff
@@ -0,0 +1,417 @@
+    testKitTest((blockchain) -> {
+      MapIndex<HashCode, TransactionLocation> txLocations = blockchain.getTxLocations();
+      Map<HashCode, TransactionLocation> txLocationsMap = toMap(txLocations);
+      TransactionLocation expectedTransactionLocation =
+          TransactionLocation.valueOf(block.getHeight(), 0L);
+      assertThat(txLocationsMap)
+          .isEqualTo(ImmutableMap.of(expectedBlockTransaction.hash(),
+              expectedTransactionLocation));
+    });
+  }
+
+  @Test
+  void getTxLocation() {
+    testKitTest((blockchain) -> {
+      Optional<TransactionLocation> txLocation =
+          blockchain.getTxLocation(expectedBlockTransaction.hash());
+      TransactionLocation expectedTransactionLocation =
+          TransactionLocation.valueOf(block.getHeight(), 0L);
+      assertThat(txLocation).hasValue(expectedTransactionLocation);
+    });
+  }
+
+  @Test
+  void getTxLocationOfUnknownTx() {
+    testKitTest((blockchain) -> {
+      HashCode unknownHash = HashCode.fromBytes(new byte[DEFAULT_HASH_SIZE_BYTES]);
+      Optional<TransactionLocation> txLocation = blockchain.getTxLocation(unknownHash);
+      assertThat(txLocation).isEmpty();
+    });
+  }
+
+  @Test
+  void getBlocks() {
+    testKitTest((blockchain) -> {
+      MapIndex<HashCode, Block> blocks = blockchain.getBlocks();
+      Map<HashCode, Block> blocksMap = toMap(blocks);
+      // Contains both genesis and committed blocks
+      assertThat(blocksMap).hasSize(2);
+      assertThat(blocksMap.get(block.getBlockHash())).isEqualTo(block);
+    });
+  }
+
+  @Test
+  void getBlockByHeight() {
+    testKitTest((blockchain) -> {
+      Block actualBlock = blockchain.getBlock(block.getHeight());
+      assertThat(actualBlock).isEqualTo(block);
+    });
+  }
+
+  @Test
+  void getBlockByInvalidHeight() {
+    testKitTest((blockchain) -> {
+      long invalidBlockHeight = block.getHeight() + 1;
+      Exception e = assertThrows(IndexOutOfBoundsException.class,
+          () -> blockchain.getBlock(invalidBlockHeight));
+      String expectedMessage = String.format("Block height (%s) is out of range [0, %s]",
+          invalidBlockHeight, block.getHeight());
+      assertThat(e).hasMessageContaining(expectedMessage);
+    });
+  }
+
+  @Test
+  void getBlockById() {
+    testKitTest((blockchain) -> {
+      Optional<Block> actualBlock = blockchain.findBlock(block.getBlockHash());
+      assertThat(actualBlock).hasValue(block);
+    });
+  }
+
+  @Test
+  void getLastBlock() {
+    testKitTest((blockchain) -> {
+      Block lastBlock = blockchain.getLastBlock();
+      assertThat(lastBlock).isEqualTo(block);
+    });
+  }
+
+  @Test
+  void getActualConfiguration() {
+    testKitTest((blockchain) -> {
+      StoredConfiguration configuration = blockchain.getActualConfiguration();
+      EmulatedNode emulatedNode = testKit.getEmulatedNode();
+      List<ValidatorKey> validatorKeys = configuration.validatorKeys();
+      short validatorCount = 1;
```

Comment:
As this test depends on a single validator network, shan't we have it as a constant, and pass it on testkit instantiation?